course3/homework-string/my_string.h

```diff
@@ -1,6 +1,60 @@
 #pragma once
+#include <vector>
+#include <iostream>
+#include <cstring>
 
 class MyString
 {
-	// TODO
+public:
+	MyString()	
+	{ 
+		// default constructor	
+	}
+
+	MyString(MyString& other)
+	{
+		this->innerStorage = other.innerStorage;
+	}
+
+	MyString(const char* initialValue)
+	{
+		innerStorage.insert(
+			innerStorage.begin(),
+			initialValue,
+			initialValue + strlen(initialValue));
+	}
+
+	bool empty() const
+	{
+		return this->size() == 0;
+	}
+
+	int size() const
+	{
+		return innerStorage.size();
+	}
+
+	bool operator==(const MyString& other)
+	{
+    	return this->innerStorage == other.innerStorage;
+	}
+
+	bool operator!=(const MyString& other)
+	{
+		return !(*this == other);
+	}
+
+	char& operator[](int index)
+	{
+		return innerStorage[index];
+	}
+	
+	char operator[](int index) const
+	{
+		return innerStorage[index];
+	}
+
+private:
+	std::vector<char> innerStorage; 
```

Comment:
This will, of course, work.
The goal of this exercise was to play with memory management. If you re-read my email, I specifically asked for explicit memory management, and to use unique_ptr<char[]>. Otherwise the task is too easy and you'll learn nothing.